File: src/main/java/com/back/pinco/domain/user/repository/UserRepository.java
package com.back.pinco.domain.user.repository;

import com.back.pinco.domain.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByEmail(String email);
    Optional<User> findByUserName(String useName);
    boolean existsByEmail(String email);
    boolean existsByUserName(String userName);
    boolean existsByUserNameAndIdNot(String userName, Long id);
    Optional<User> findByApiKey(String apiKey);
    boolean existsByApiKey(String key);
}

File: src/main/java/com/back/pinco/domain/user/repository/UserRepository.kt
package com.back.pinco.domain.user.repository

import com.back.pinco.domain.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findByEmail(email: String): User?
    fun findByUserName(userName: String): User?
    fun existsByEmail(email: String): Boolean
    fun existsByUserName(userName: String): Boolean
    fun existsByUserNameAndIdNot(userName: String, id: Long): Boolean
    fun findByApiKey(apiKey: String): User?
    fun existsByApiKey(key: String): Boolean
}
